### ISSUE TITLE
Needs install for easier acquisition

### ISSUE THREAD

**Issue by heaths** [0xd4d/dnSpy#1149]
Currently, the process for acquiring dnSpy requires several steps:

1. Figure out which zip to download (granted, not hard).
2. Unzip to a directory.
3. Unblock every file because of "Mark of the Internet" (fortunately, PowerShell makes this easy for people who know about `Unblock-File`).
4. Copy (if you haven't already extracted) to some directory.
5. (Optional) Create a shortcut.

Having an installer for this would be great. As a long-time setup developer (did a lot of work on WiX, beta-tested MSI, worked on VS setup for years, [vswhere](https://github.com/microsoft/vswhere), etc., I'd love to offer help but wanted to start a conversation on how you'd like acquisition. ILSpy, for example, supports acquisition via [chocolatey](https://chocolatey.org). Could always have an MSI-based installer via WiX. I've developed a lot of repos that do both and more, so if you have ideas what you'd want (if anything) I'd love to help. After learning of this, I'm liking it over ILSpy.

**Comment by 0xd4d** [comment]
I don't have to unblock any file, which browser did you use?

I like the no-installer zip files, nothing to uninstall just delete the directory when you don't need it anymore. It's portable too.

**Comment by 0xd4d** [comment]
Actually it could be useful to have an installer in addition to the zip files since the .NET Core zip files have 421 files + dirs in them so it's difficult to find the main exe. If you want to do it, send a PR. There's also a netcore3.0 branch that I'll merge into master at the latest when .NET Core 3.0 is released.

**Comment by heaths** [comment]
I'm using Chredge/Edgium. Happens on Edge as well, but last I checked even Chrome was setting Mark of the Internet on ZIPs. What might be different is that I just use Windows Explorer's unzip, which transfers the MOTI to files it extracts (which is by design). Other programs may not, but I've never had need of separate unzippers (just more to configure on a newly formatted system).

**Comment by Styxxy** [comment]
Might be interesting to use msix and distribute through the windows store. You can even create different channels. A release pipeline isn't too complex to setup (and free with azure devops).

You could get inspiration from https://github.com/NuGetPackageExplorer/NuGetPackageExplorer .

**Comment by DotNetSparky** [comment]
Just as an aside (re: unblocking the files):  unblocking the zip file before you unzip will help avoid that (admittedly I forget to do that more times than not and usually delete and unzip again, so thank-you for the powershell tip 🙂).

**Comment by heaths** [comment]
Yeah, but same here: forgot until after it was unzipped. 😒 I've got plans this weekend but hope to get a start on this next week. Will use self-contained MSIs: one for x86-only, and one for our with x86 since that's needed to debug x86 apps. Will see about sharing Core binaries to reduce size and rename x86 host binary (your docs even mention this, though extracted it still is called dnSpy.exe). Can also do authoring to publish for chocolatey if you want, but you'll need to secure a PAT. I'd just use placeholders for variables, of course.

**Comment by 0xd4d** [comment]
@heaths Just install .NET Framework x86 and x64 files to `Program Files (x86)` and the .NET Framework files only need one installer, just like it only has one zip file today.

**Comment by glenn-slayden** [comment]
for best/paranoid security, i now habitually unzip as follows on every/any download from anywhere:
- right click properties on the downloaded zip
- click “unblock”, press ‘apply’ (not “close” yet)
- also check the read-only box, THEN close.
- now, run your virus scan specifically on the read only zip file
- only now, use 7z to unzip into a folder...
